scripts/process/trends_all.R

```diff
@@ -0,0 +1,99 @@
+process.trends_all <- function(viz=as.viz("trends_all")){
+  library(dplyr)
+  library(tidyr)
+  library(rkt)
+  library(lubridate)
+
+  # load data
+  deps <- readDepends(viz)
+  sessions_all <- deps[["sessions_all"]]
+  year_filter <- deps[["year_filter"]]
+  project_table <- deps[['project_table']]
+
+  # summarize the year's session records into daily session counts for each viewID
+  counts_df <- year_filter %>%
+    group_by(viewID, date) %>%
+    summarize(sessions=sum(sessions, na.rm=TRUE)) %>%
+    ungroup()
+
+  # declare the time windows we want to analyze
+  range_text <- c("-1 year","-1 month","-1 week")
+  names(range_text) <- c("Year","Month","Week")
+  latest_day <- max(counts_df$date, na.rm = TRUE)
+
+  # pad counts with 0's on days when no sessions are reported
+  year_range <- seq(latest_day, length = 2, by = range_text["Year"])
+  seq_days <- seq(year_range[2], year_range[1], by=as.difftime(1, units="days"))
+  counts_df_pad <- counts_df %>%
+    bind_rows(data_frame(viewID=NA, date=seq_days)) %>%
+    spread(date, sessions, fill=0) %>%
+    gather(date, sessions, -viewID) %>%
+    filter(!is.na(viewID)) %>% # remove the dummy rows that were just to ensure every date got included
+    mutate(date = as.Date(date)) # restore date format that got lost when
+
+  # augment counts with columns that will help us compute trends
+  if(any(is.na(counts_df_pad$sessions))) {
+    stop("oops, Alison thought we'd never see NAs. better add some na.rms in process.trends_all()")
+  }
+  counts_df_aug <- counts_df_pad %>%
+    group_by(viewID) %>%
+    mutate(
+      sessions_mean = mean(sessions),
+      sessions_norm = sessions / mean(sessions),
+      dec_date = lubridate::decimal_date(date),
+      day_of_week = lubridate::wday(date)
+    ) %>%
+    ungroup()
+
+  # create 1-year, 1-month, and 1-week subsets
+  counts <- list()
```

Comment:
This calc is done for the timeline graphs as well. I'm going to merge this, and shuffle around the code so we can optimize it a bit. 

I'm a little worried we're losing the factors that are so important in the viz step, but the actual trend calculations here are what we need